Helper/AutomatedTesting.php

```diff
@@ -0,0 +1,333 @@
+        $this->shippingMethodConverter = $shippingMethodConverter;
+        $this->bugsnag = $bugsnag;
+        $this->storeManager = $storeManager;
+        $this->quoteFactory = $quoteFactory;
+        $this->quoteManagement = $quoteManagement;
+        $this->quoteRepository = $quoteRepository;
+    }
+
+    /**
+     * Generate and return automated testing config
+     *
+     * @return AutomatedTestingConfig|string
+     */
+    public function getAutomatedTestingConfig()
+    {
+        try {
+            $simpleProduct = $this->getProduct(\Magento\Catalog\Model\Product\Type::TYPE_SIMPLE);
+            if ($simpleProduct === null) {
+                return 'no simple products found';
+            }
+
+            $virtualProduct = $this->getProduct(\Magento\Catalog\Model\Product\Type::TYPE_VIRTUAL);
+            $simpleStoreItem = $this->convertToStoreItem($simpleProduct, 'simple');
+            $virtualStoreItem = $this->convertToStoreItem($virtualProduct, 'virtual');
+            $storeItems[] = $simpleStoreItem;
+            if ($virtualStoreItem !== null) {
+                $storeItems[] = $virtualStoreItem;
+            }
+
+            $quote = $this->createQuoteWithItem($simpleProduct);
+            $shippingMethods = $this->getShippingMethods($quote);
+            if (empty($shippingMethods)) {
+                return 'no shipping methods found';
+            }
+
+            $quote->collectTotals();
+            $this->quoteRepository->save($quote);
+            $simpleCartItem = $this->cartItemFactory->create()
+                                                    ->setName($simpleStoreItem->getName())
+                                                    ->setPrice($simpleStoreItem->getPrice())
+                                                    ->setQuantity(1);
+            $cart = $this->cartFactory->create()
+                                      ->setItems([$simpleCartItem])
+                                      ->setShipping(reset($shippingMethods))
+                                      ->setExpectedShippingMethods($shippingMethods)
+                                      ->setTax($this->formatPrice($quote->getShippingAddress()->getTaxAmount()))
+                                      ->setSubTotal($this->formatPrice($quote->getSubtotal()));
+            $this->quoteRepository->delete($quote);
+
+            return $this->configFactory->create()->setStoreItems($storeItems)->setCart($cart);
+        } catch (\Exception $e) {
+            $this->bugsnag->notifyException($e);
+            return 'error retrieving automated testing config';
+        }
+    }
+
+    /**
+     * Return a product with the specified type
+     *
+     * @param string $type
+     *
+     * @return Product|null
+     */
+    protected function getProduct($type)
+    {
+        $this->sortOrder->setField('price')->setDirection('DESC');
+
+        $searchCriteria = $this->searchCriteriaBuilder
```

Comment:
need to find product with inventory (I think in another PR)

Reply:
added. its a small addition.